server/admin/users.js

```diff
@@ -0,0 +1,102 @@
+const router = require('express').Router()
+const { User } = require('../db/models')
+const { adminMiddleware } = require('../middleware')
+
+module.exports = router
+
+router.get('/', adminMiddleware, async (req, res, next) => {
+  try {
+    const users = await User.findAll({
+      attributes: [
+        'id',
+        'email',
+        'firstName',
+        'lastName',
+        'isAdmin',
+        'emailOptIn'
+      ]
+    })
+    res.json(users)
+  } catch (err) {
+    next(err)
+  }
+})
+
+router.get('/:userId', adminMiddleware, async (req, res, next) => {
+  try {
+    const user = await User.findByPk(req.params.userId)
+    user ? res.json(user) : res.status(400).end()
+  } catch (err) {
+    next(err)
+  }
+})
+
+router.post('/', adminMiddleware, async (req, res, next) => {
+  try {
+    const {
+      firstName,
+      lastName,
+      email,
+      password,
+      googleId,
+      isAdmin,
+      emailOptIn
+    } = req.body
+    const newUser = await User.create({
+      firstName,
+      lastName,
+      email,
+      password,
+      googleId,
+      isAdmin,
+      emailOptIn
+    })
+    newUser ? res.json(newUser) : res.status(400).end()
+  } catch (err) {
+    next(err)
+  }
+})
+
+router.put('/:userId', adminMiddleware, async (req, res, next) => {
+  try {
+    const {
+      firstName,
+      lastName,
+      email,
+      password,
```

Comment:
should we allow admin to update password? I feel like the user should be the only one... unless we can reset the password